src/common/updating/http_client.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <future>
+#include <winrt/Windows.Web.Http.h>
+
+namespace http
+{
+    class HttpClient
+    {
+    public:
+        HttpClient();
+        std::future<std::wstring> request(const winrt::Windows::Foundation::Uri& url);
+        std::future<void> download(const winrt::Windows::Foundation::Uri& url, const std::wstring& dstFle);
+        std::future<void> download(const winrt::Windows::Foundation::Uri& url, const std::wstring& dstFle, const std::function<void(float)>& progressUpdateHandle);
+
+    private:
+        std::future<void> downloadFile(uint64_t totalDownloadSize, const winrt::Windows::Storage::Streams::IInputStream& contentStream, const std::wstring& dstFilePath);

```

Comment:
`downloadFile` makes `download` methods naming confusing. perhaps make it a free function or embed it to `download`, since it's used only in one place? that way we could also reduce mutable state by removing `m_progressChangeHandle` member altogether.